src/hierarchy.ts

```diff
@@ -74,17 +74,14 @@ export class Group implements AsyncMutableMapping<Group | ZarrArray> {
 
     private static async loadMetadataForConstructor(store: Store, path: null | string) {
         path = normalizeStoragePath(path);
-
-        if (await containsArray(store, path)) {
-            throw new ContainsArrayError(path);
-        }
-
         const keyPrefix = pathToPrefix(path);
         try {
             const metaStoreValue = await store.getItem(keyPrefix + GROUP_META_KEY);
             return parseMetadata(metaStoreValue);
-        }
-        catch (error) {
+        } catch (error) {
+            if (await containsArray(store, path)) {
+                throw new ContainsArrayError(path);
+            }
```

Comment:
Might be a redundant check, but it's within the catch block and only executed if the group metadata doesn't exist.